x-pack/legacy/plugins/infra/public/components/waffle/waffle_inventory_switcher.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiPopover, EuiContextMenu, EuiButton } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import React, { useCallback, useState, useMemo } from 'react';
+import {
+  InfraSnapshotMetricInput,
+  InfraSnapshotMetricType,
+  InfraNodeType,
+  InfraSnapshotGroupbyInput,
+} from '../../graphql/types';
+
+interface Props {
+  nodeType: InfraNodeType;
+  changeNodeType: (nodeType: InfraNodeType) => void;
+  changeGroupBy: (groupBy: InfraSnapshotGroupbyInput[]) => void;
+  changeMetric: (metric: InfraSnapshotMetricInput) => void;
+}
+
+export const WaffleInventorySwitcher = (props: Props) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const closePopover = useCallback(() => setIsOpen(false), []);
+  const openPopover = useCallback(() => setIsOpen(true), []);
+  const goToNodeType = useCallback(
+    (nodeType: InfraNodeType) => {
+      closePopover();
+      props.changeNodeType(nodeType);
+      props.changeGroupBy([]);
+      props.changeMetric({ type: InfraSnapshotMetricType.cpu });
+    },
+    [props.changeGroupBy, props.changeNodeType, props.changeMetric]
+  );
+  const goToHost = useCallback(() => goToNodeType('host' as InfraNodeType), [goToNodeType]);
+  const goToK8 = useCallback(() => goToNodeType('pod' as InfraNodeType), [goToNodeType]);
+  const goToDocker = useCallback(() => goToNodeType('container' as InfraNodeType), [goToNodeType]);
+  const panels = useMemo(
+    () => [
+      {
+        id: 0,
+        items: [
+          {
+            name: i18n.translate('xpack.infra.waffle.nodeTypeSwitcher.hostsLabel', {
+              defaultMessage: 'Hosts',
+            }),
+            icon: 'host',
+            onClick: goToHost,
+          },
+          {
+            name: 'Kubernets',
+            icon: 'kubernets',
+            onClick: goToK8,
+          },
+          {
+            name: 'Docker',
+            icon: 'docker',
+            onClick: goToDocker,
+          },
+        ],
+      },
+    ],
+    []
+  );
+  const selectedText = useMemo(() => {
+    switch (props.nodeType) {
+      case InfraNodeType.host:
+        return i18n.translate('xpack.infra.waffle.nodeTypeSwitcher.hostsLabel', {
+          defaultMessage: 'Hosts',
+        });
+      case InfraNodeType.pod:
+        return 'Kubernetes';
+      case InfraNodeType.container:
+        return 'Docker';
+    }
+  }, [props.nodeType]);
+
+  return (
+    <EuiPopover
+      id="contextMenu"
+      button={
+        <EuiButton iconType="arrowDown" iconSide="right" onClick={openPopover}>
+          {selectedText}
+        </EuiButton>
```

Comment:
This should be an `EUIFilterButton` similar to how the other drop downs look. @hbharding Thoughts

Reply:
Hey @simianhacker - thanks for catching. So Yes and no. It turns out that `EuiFilterButton` doesn't take a `prepend` prop for adding a label... however I mentioned this to the EUI team and they've added it for discussion here https://github.com/elastic/eui/issues/2286. In the meantime, I see two options:
1) We build this ourselves with some custom html/css
2) We do what we are currently doing - add the "label" to the button text itself. i.e. `Metric: CPU Usage`

Reply:
I vote for option 2. Adding the label would clear things up, and don't want to end up with what will undoubtly be a worse solution than EUI is already working on.

Reply:
Option 2 is fine but the buttons need to all have the same styling so either `EuiButton` or `EuiFilterButton`. I don't mind if we create a `MetricsUIToolbarButton` that wraps up either and customizes it some. Then later, once EUI catches up, we can just fix it in that one file.